docs/password_hashing.rst

```diff
@@ -0,0 +1,136 @@
+.. _password-hashing:
+
+Password hashing
+================
+
+.. currentmodule:: nacl.pw_hash
+
+An ever important use of cryptographic hashing primitives has been
```

Comment:
Let's remove this entire first paragraph as it isn't conveying useful information.